modules/mysql/mysql.go

```diff
@@ -0,0 +1,524 @@
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/netdata/go.d.plugin/modules"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func init() {
+	creator := modules.Creator{
+		Create: func() modules.Module { return New() },
+	}
+
+	modules.Register("mysql", creator)
+}
+
+// MySQL is the mysql database module.
+type MySQL struct {
+	modules.Base
+	db *sql.DB
+	// i.e user:password@/dbname
+	DSN string `yaml:"dsn"`
+}
+
+// New creates and returns a new empty MySQL module.
+func New() *MySQL {
+	// return &MySQL{SlowQueriesMaxCount: 0, SlowQueriesInPercentile: 95, SlowQueriesSeconds: 4}
+	return &MySQL{}
+}
+
+// CompatibleMinimumVersion is the minimum required version of the mysql server.
+const CompatibleMinimumVersion = 5.1
+
+func (m *MySQL) getMySQLVersion() float64 {
+	var versionStr string
+	var versionNum float64
+	if err := m.db.QueryRow("SELECT @@version").Scan(&versionStr); err == nil {
+		versionNum, _ = strconv.ParseFloat(regexp.MustCompile(`^\d+\.\d+`).FindString(versionStr), 64)
+	}
+
+	return versionNum
+}
+
+// Cleanup performs cleanup.
+func (m *MySQL) Cleanup() {
+	err := m.db.Close()
+	if err != nil {
+		m.Errorf("cleanup: error on closing the mysql database [%s]: %v", m.DSN, err)
+	}
+}
+
+// Init makes initialization of the MySQL mod.
+func (m *MySQL) Init() bool {
+	if m.DSN == "" {
+		m.Warningf("'DSN' field is missing, trying to connect through default socket...")
+		if !m.tryResolveDSN() {
+			m.Errorf("couldn't connect, 'DSN' field is missing and connection through default DSN failed")
+			return false
+		}
+	}
+
+	// test the connectivity here.
+	if err := m.openConnection(); err != nil {
+		return false
+	}
+
+	if min, got := CompatibleMinimumVersion, m.getMySQLVersion(); min > 0 && got < min {
+		m.Warningf("running with uncompatible mysql version [%v<%v]", got, min)
```

Comment:
No, the idea was that, a metrics collector which has a `Version()` method would run through a loop inside the main module, compare `getCurrentMySQLVersion()` version vs collector's `Version()` which should be >= in order to that collect to run, otherwise a warning would show up.